src/ConfigManagerHelper.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_content;
+
+use Drupal\Core\Config\ConfigManagerInterface;
+use Drupal\Core\Config\StorageInterface;
+
+/**
+ * The ConfigManagerHelper class.
+ *
+ * It provides additional methods to work with the configuration system.
+ */
+class ConfigManagerHelper implements ConfigManagerHelperInterface {
+
+  /**
+   * Configuration manager.
+   *
+   * @var \Drupal\Core\Config\ConfigManagerInterface
+   */
+  protected $configManager;
+
+  /**
+   * Entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * ConfigurationManagerHelper constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigManagerInterface $config_manager
+   *   Config manager.
+   */
+  public function __construct(ConfigManagerInterface $config_manager) {
+    $this->configManager = $config_manager;
+    $this->entityTypeManager = $config_manager->getEntityTypeManager();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createConfig(string $config_name, StorageInterface $storage): void {
```

Comment:
This assumes that all config are config entities, while some config files are simple config object, and this method won't work in that case.

We already have a core service that we can use to create new configuration, and it works with both config entities and simple config object. Check `\Drupal\Core\Config\ConfigInstaller::installOptionalConfig()` the service is called `'config.installer'`. We will basically point this method to the post update config directory that contains the config snapshot (see my previous comment).

Reply:
We could use this issue to investigate the service right? https://citnet.tech.ec.europa.eu/CITnet/jira/browse/D8TEWPPE-53